pimd/pim_cmd_common.c

```diff
@@ -653,3 +673,1302 @@ int pim_process_no_rp_plist_cmd(struct vty *vty, const char *rp_str,
+				vty_out(vty,
+					"%pPAs : %pPAs is a source, uptime is %s\n",
+					&up->sg.grp, &up->sg.src, uptime);
+			}
+
+			if (!print_header) {
+				vty_out(vty, "\n");
+				vty_out(vty, "\n");
+			}
+
+			vty_out(vty, "Hellos\n");
+			vty_out(vty, "------\n");
+			vty_out(vty, "Period         : %d\n",
+				pim_ifp->pim_hello_period);
+			vty_out(vty, "HoldTime       : %d\n",
+				PIM_IF_DEFAULT_HOLDTIME(pim_ifp));
+			vty_out(vty, "Timer          : %s\n", hello_timer);
+			vty_out(vty, "StatStart      : %s\n", stat_uptime);
+			vty_out(vty, "Receive        : %d\n",
+				pim_ifp->pim_ifstat_hello_recv);
+			vty_out(vty, "Receive Failed : %d\n",
+				pim_ifp->pim_ifstat_hello_recvfail);
+			vty_out(vty, "Send           : %d\n",
+				pim_ifp->pim_ifstat_hello_sent);
+			vty_out(vty, "Send Failed    : %d\n",
+				pim_ifp->pim_ifstat_hello_sendfail);
+			vty_out(vty, "Generation ID  : %08x\n",
+				pim_ifp->pim_generation_id);
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+
+			pim_print_ifp_flags(vty, ifp);
+
+			vty_out(vty, "Join Prune Interval\n");
+			vty_out(vty, "-------------------\n");
+			vty_out(vty, "LAN Delay                    : %s\n",
+				pim_if_lan_delay_enabled(ifp) ? "yes" : "no");
+			vty_out(vty, "Effective Propagation Delay  : %d msec\n",
+				pim_if_effective_propagation_delay_msec(ifp));
+			vty_out(vty, "Effective Override Interval  : %d msec\n",
+				pim_if_effective_override_interval_msec(ifp));
+			vty_out(vty, "Join Prune Override Interval : %d msec\n",
+				pim_if_jp_override_interval_msec(ifp));
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+
+			vty_out(vty, "LAN Prune Delay\n");
+			vty_out(vty, "---------------\n");
+			vty_out(vty, "Propagation Delay           : %d msec\n",
+				pim_ifp->pim_propagation_delay_msec);
+			vty_out(vty, "Propagation Delay (Highest) : %d msec\n",
+				pim_ifp->pim_neighbors_highest_propagation_delay_msec);
+			vty_out(vty, "Override Interval           : %d msec\n",
+				pim_ifp->pim_override_interval_msec);
+			vty_out(vty, "Override Interval (Highest) : %d msec\n",
+				pim_ifp->pim_neighbors_highest_override_interval_msec);
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+
+			vty_out(vty, "BSM Status\n");
+			vty_out(vty, "----------\n");
+			vty_out(vty, "Bsm Enabled          : %s\n",
+				pim_ifp->bsm_enable ? "yes" : "no");
+			vty_out(vty, "Unicast Bsm Enabled  : %s\n",
+				pim_ifp->ucast_bsm_accept ? "yes" : "no");
+			vty_out(vty, "\n");
+			vty_out(vty, "\n");
+		}
+	}
+
+	if (!found_ifname)
+		vty_out(vty, "%% No such interface\n");
+}
+
+void ip_pim_ssm_show_group_range(struct pim_instance *pim, struct vty *vty,
+				 bool uj)
+{
+	struct pim_ssm *ssm = pim->ssm_info;
+	const char *range_str =
+		ssm->plist_name ? ssm->plist_name : PIM_SSM_STANDARD_RANGE;
+	json_object *json;
+
+	if (uj) {
+		json = json_object_new_object();
+		json_object_string_add(json, "ssmGroups", range_str);
+	} else
+		vty_out(vty, "SSM group range : %s\n", range_str);
+
+#if PIM_IPV == 6
```

Comment:
condition reversed, i think this is ipv4?

Reply:
Range will be FF3X::/32 only for ipv6, hence the PIM_IPV == 6 condition

Reply:
Okay, but `ff3x::/32` isn't valid input for an IPv6 prefix, and `strcmp` is not the right tool to compare a prefix…